src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -118,8 +117,7 @@ public virtual void Init(IProviderConfiguration providerCfg, string providerName
             serviceProvider = svcProvider;
             receivers = new ConcurrentDictionary<QueueId, EventHubAdapterReceiver>();
             this.SerializationManager = this.serviceProvider.GetRequiredService<SerializationManager>();
-            this.getNodeConfig = svcProvider.GetRequiredService<Func<NodeConfiguration>>();
-
+            this.getNodeConfig = new Lazy<Func<NodeConfiguration>>(() => this.serviceProvider.GetRequiredService<Func<NodeConfiguration>>());
```

Comment:
+1. sorry I didn't put on some context on the description, which makes the PR very confusing